tests/test_binning.py

```diff
@@ -334,4 +340,186 @@ def test_init(self):
 
     def test_determine_total_bins(self):
         mab = MABBinMapper([5])
+        # Test bin counting
         assert mab.determine_total_bins(nbins_per_dim=[5, 1], direction=[1, 86], skip=[0, 0], bottleneck=True) == 9
+        assert mab.determine_total_bins(nbins_per_dim=[5, 5], direction=[0, 0], skip=[0, 0], bottleneck=True) == 33
+        assert mab.determine_total_bins(nbins_per_dim=[5, 5, 5], direction=[0, 0, 0], skip=[0, 0, 0], bottleneck=True) == 137
+        assert mab.determine_total_bins(nbins_per_dim=[5, 5], direction=[0, 0], skip=[1, 0], bottleneck=True) == 9
+        assert mab.determine_total_bins(nbins_per_dim=[5, 1], direction=[1, 86], skip=[0, 0], bottleneck=False) == 6
+
+        # Test bin assignments with 2D and 3D coords
+        # First load reference datasets from file
+        test_refs = []
+        with h5py.File(os.path.join(REFERENCE_PATH, 'mab_assignments_ref.h5'), 'r') as f:
```

Comment:
The ref file is not included in this PR.

Reply:
Sorry it got blocked by gitignore. I added it in a subsequent commit.